Release automation failed for version 3.0.1

https://github.com/asciidoctor/asciidoctor-vscode/actions/runs/2677642232
```
 DONE  Published asciidoctor.asciidoctor-vscode v3.0.1.
To https://github.com/asciidoctor/asciidoctor-vscode
   f3eb0ce..a69e4c9  master -> master
https://github.com/asciidoctor/asciidoctor-vscode/releases/tag/untagged-870318b8d420cdeed2a0
release not found
## master...origin/master
 M CHANGELOG.md
?? release-notes.md
Error: Process completed with exit code 1.
```

In addition, the previous version (3.0.0) was missing from the full changelog link.

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
 	"name": "asciidoctor-vscode",
 	"displayName": "%displayName%",
 	"description": "%description%",
-	"version": "3.0.1",
+	"version": "3.0.1-dev",
 	"publisher": "asciidoctor",
 	"author": "João Pinto <lamego.pinto@gmail.com>",
 	"license": "MIT",

--- a/release.sh
+++ b/release.sh
@@ -27,13 +27,16 @@ git config --local user.email "$RELEASE_GIT_EMAIL"
   # > Support for this will arrive in the future.
   # https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions
   RELEASE_VERSION_WITHOUT_PRERELEASE=$(node -e "console.log(require('semver').coerce('$RELEASE_VERSION').version)")
+  PRERELEASE_VERSION=$(node -e "console.log(require('semver').parse('$RELEASE_VERSION').prerelease.length > 0)")
   npm version $RELEASE_VERSION_WITHOUT_PRERELEASE --message "release $RELEASE_VERSION_WITHOUT_PRERELEASE [no ci]"
   git push origin $(git describe --tags --exact-match)
   npm run package
-  npx vsce publish -p $RELEASE_VSCE_TOKEN $(node -e "console.log(require('semver').parse('$RELEASE_VERSION').prerelease.length > 0 ? '--pre-release' : '')")
+  RELEASE_VSCE_PRERELEASE_OPT=$([[ "$PRERELEASE_VERSION" == "true" ]] && echo "--pre-release" || echo "")
+  npx vsce publish -p $RELEASE_VSCE_TOKEN $RELEASE_VSCE_PRERELEASE_OPT
   git push origin $RELEASE_BRANCH
   node tasks/release-notes.js
-  gh release create v$RELEASE_VERSION_WITHOUT_PRERELEASE -t v$RELEASE_VERSION_WITHOUT_PRERELEASE -F release-notes.md -d
+  RELEASE_GH_PRERELEASE_OPT=$([[ "$PRERELEASE_VERSION" == "true" ]] && echo "--prerelease" || echo "")
+  gh release create v$RELEASE_VERSION_WITHOUT_PRERELEASE -t v$RELEASE_VERSION_WITHOUT_PRERELEASE -F release-notes.md $RELEASE_GH_PRERELEASE_OPT
   gh release upload v$RELEASE_VERSION_WITHOUT_PRERELEASE asciidoctor-vscode-$RELEASE_VERSION_WITHOUT_PRERELEASE.vsix
   npm pkg set version="$(npx semver -i patch $RELEASE_VERSION_WITHOUT_PRERELEASE)-dev"
   git commit -a -m 'prepare branch for development [no ci]'

--- a/tasks/release-notes.js
+++ b/tasks/release-notes.js
@@ -53,12 +53,10 @@ async function getReleaseChangelog() {
 }
 
 ;(async () => {
-  const previousTag = await execute('sh', ['-c', 'git tag -l --sort -taggerdate | head -n2 | tail -n +2'])
+  const previousTag = await execute('sh', ['-c', 'git -c versionsort.suffix=. -c versionsort.suffix=- ls-remote --tags --refs --sort -v:refname origin | head -n+2 | tail -n +2 | cut -d" " -f2 | cut -d"/" -f3'])
   const releaseChangelog = await getReleaseChangelog()
 
   const notes = `## What's Changed
-
-Write summary...
 
 ## Changelog
 ${releaseChangelog}

